src/Symfony/Component/DependencyInjection/Compiler/MergeExtensionConfigurationPass.php

```diff
@@ -117,6 +118,15 @@ public function __construct(parent $parameterBag)
         $this->mergeEnvPlaceholders($parameterBag);
     }
 
+    public function resolveValue(mixed $value, array $resolving = []): mixed
+    {
+        if ($value instanceof ParamConfigurator) {
```

Comment:
`ParamConfigurator` and `EnvConfigurator` are special because of the "placeholder" stuff (see BaseNode).

Reply:
which test fails without this override?

Reply:
No test fail but `ParamConfigurator` and `EnvConfigurator` doesn't work as expected because their string value won't be an env placeholder string when they are treated in BaseNode, but still an object.